doc/source/user_guide/codelist.rst

```diff
@@ -0,0 +1,244 @@
+.. _codelist:
+
+Validation using Codelists
+==========================
+
+A codelist is a list of codes (i.e. allowed values). In this package, the codelists are
+parsed from yaml files. They contain lists of allowed values for any **index**
+dimension of the IAMC format (model, scenario, region, variable). Which index dimension
+a codelist will be applied to is determined by the **name of the folder** in which it is
+located. For example, if a codelist is located in a folder called ``model/`` it will be
+applied to the "model" index dimension. 
+
+Codelist format specification
+-----------------------------
+
+In the most simple case of a codelist it contains a list of strings, e.g.:
+
+.. code:: yaml
+
+  - allowed_value_a
+  - allowed_value_b
+  - allowed_value_c
+  - ...
+
+
+For defining allowed *variables* and *regions*, the list elements are more complicated
+than simple strings, detailed in the following.
+
+.. _variable:
+
+Variable
+^^^^^^^^
+
+An entry in a variable code list, *must be* a mapping (translated to python as a
+dictionary). It maps the **name** of an allowed variable to, at least, one key-value
+pair defining the allowed **unit(s)** for the variable.
+
+This is an example for a valid entry in a variable codelist:
+
+.. code:: yaml
+
+   - Allowed variable name:
+     unit: A unit
+     description: A short description
+     <other attribute>: Some text, value, boolean or list (optional)
+
+The **unit** attribute is **required** and its value should be compatible with the
+`iam-units <https://github.com/iamconsortium/units>`_ package.
+
+The unit attribute can be:
```

Comment:
This is not formatted as a list on RTD...

Reply:
Added a second line break, now it's rendered correctly.